front-end/src/app/reports/f1m/main-form/main-form.component.html

```diff
@@ -251,6 +251,12 @@ <h2>Affiliation</h2>
             <div class="field">
               <label for="name">AFFILIATED DATE</label>
               <p-calendar
+                (onFocus)="calendarOpened = true"
+                (onClose)="
+                  calendarOpened = false;
```

Comment:
could you make this an onClose method in the component where the form field name is the argument?

Reply:
I updated onClose and onBlur to be a function which would move the SchemaUtils.onBlurValidation() function call to the ts instead of the html